filebeat/input/filestream/internal/input-logfile/harvester.go

```diff
@@ -217,6 +219,36 @@ func startHarvester(ctx input.Context, hg *defaultHarvesterGroup, s Source, rest
 	}
 }
 
+// Continue start a new Harvester with the state information from a different Source.
+func (hg *defaultHarvesterGroup) Continue(ctx input.Context, previous, next Source) {
+	ctx.Logger.Debugf("Continue harvester for file prev=%s, next=%s", previous.Name(), next.Name())
+	prevID := hg.identifier.ID(previous)
+	nextID := hg.identifier.ID(next)
+
+	hg.tg.Go(func(canceler unison.Canceler) error {
+		previousResource, err := lock(ctx, hg.store, prevID)
+		if err != nil {
+			return fmt.Errorf("error while locking previous resource: %v", err)
+		}
+		// mark previous state out of date
+		// so when reading starts again the offset is set to zero
+		hg.store.UpdateTTL(previousResource, -1)
```

Comment:
What impact has the `-1` have here? Will the entry be gc'ed on the next round? As it is gone, should we try to remove right away or mark as 'gone', so we don't pick up an old state on inode reuse by accident?

Reply:
This is a mistake. I wanted to remove it as soon as possible.